code/modules/vore/fluffstuff/custom_items_yw.dm

```diff
@@ -538,4 +538,35 @@
 	. = ..()
 	if (M == user)
 		M.resize(3)
-		M.drop_from_inventory(src)
\ No newline at end of file
+		M.drop_from_inventory(src)
+
+// //////////////////// //
+// Telescoping Unbrella //
+// //////////////////// //
+
+/obj/item/weapon/melee/umbrella/telescoping
+	name = "telescoping umbrella"
+	desc = "A compack telescoping umbrella,use with caution on windy days.."
+	icon = 'icons/obj/items.dmi'
+	icon_state = "umbrella_closed"
+	addblends = "umbrella_closed_a"
+	slot_flags = SLOT_BELT
+	force = 2 //Its smaller
+	throwforce = 3
+
+/obj/item/weapon/melee/umbrella/telescoping/proc/extend_toggle_umbrella()
+	open = !open
+	icon_state = "umbrella_[open ? "open" : "closed"]"
+	addblends = icon_state + "_a"
+	item_state = icon_state
+	update_icon()
+	if(ishuman(src.loc))
+		var/mob/living/carbon/human/H = src.loc
+		H.update_inv_l_hand(0)
+		H.update_inv_r_hand()
```

Comment:
Instead of copy-pasting the code from umbrellas, why not just override the toggle proc with the size changing?

Reply:
This would also fix the "not working" bit.

Reply:
Oh as in do a ..()
then have my code?